libs/hooks/open-telemetry/src/lib/open-telemetry-hook.ts

```diff
@@ -1,65 +1,37 @@
-import {
-  Hook,
-  HookContext,
-  EvaluationDetails,
-  FlagValue,
-} from '@openfeature/js-sdk';
-import { Span, Tracer, trace } from '@opentelemetry/api';
+import { Hook, HookContext, EvaluationDetails, FlagValue } from '@openfeature/js-sdk';
+import { trace } from '@opentelemetry/api';
 
-const SpanProperties = Object.freeze({
-  FLAG_KEY: 'feature_flag.flag_key',
+const eventName = 'feature_flag';
+const SpanEventProperties = Object.freeze({
+  FLAG_KEY: 'feature_flag.key',
   PROVIDER_NAME: 'feature_flag.provider_name',
-  VARIANT: 'feature_flag.evaluated_variant',
-  VALUE: 'feature_flag.evaluated_value',
+  VARIANT: 'feature_flag.variant',
 });
 
 export class OpenTelemetryHook implements Hook {
-  private spanMap = new WeakMap<HookContext, Span>();
-  private tracer: Tracer;
-
-  constructor() {
-    this.tracer = trace.getTracer(
-      '@openfeature/open-telemetry-hook',
-      '5.0.0' // x-release-please-version
-    );
-  }
-
-  before(hookContext: HookContext) {
-    const span = this.tracer.startSpan(
-      `${hookContext.providerMetadata.name} ${hookContext.flagKey}`,
-      {
-        attributes: {
-          [SpanProperties.FLAG_KEY]: hookContext.flagKey,
-          [SpanProperties.PROVIDER_NAME]: hookContext.providerMetadata.name,
-        },
+  after(hookContext: HookContext, evaluationDetails: EvaluationDetails<FlagValue>) {
+    const currentTrace = trace.getActiveSpan();
+    if (currentTrace) {
+      let variant = evaluationDetails.variant;
+
+      if (!variant) {
+        if (typeof evaluationDetails.value === 'string') {
+          variant = evaluationDetails.value;
+        } else {
+          variant = JSON.stringify(evaluationDetails.value);
```

Comment:
You could refactor this into 2 ternaries, which would allow you to make `variant` a const, also... but some people hate nested ternaries.

Reply:
Yeah, I had a nested ternary but that seemed a bit nasty too. Can you provide a code snippet of what you had in mind?

Reply:
Something like:

```typescript
const variant = (evaluationDetails.variant) ?
        evaluationDetails.variant : (typeof evaluationDetails.value === 'string') ? 
        evaluationDetails.value : JSON.stringify(evaluationDetails.value);
```

Reply:
Some of the braces here aren't needed, but I find they make nested ternaries easier to read.

Again, some people HATE them though.

Reply:
I'm just going to leave it. Prettier makes your proposal look nasty, unfortunately. Also, SonarLint is also complaining about nested ternaries.